service/controller/clusterapi/v29/resource/tcdp/create.go

```diff
@@ -120,22 +156,200 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-//func (r *Resource) newIAMRolesParams(ctx context.Context, cr v1alpha1.Cluster) (*template.ParamsMainIAMRoles, error) {
-//	cc, err := controllercontext.FromContext(ctx)
-//	if err != nil {
-//		return nil, microerror.Mask(err)
-//	}
+func (r *Resource) newAutoScalingGroup(ctx context.Context, cl v1alpha1.Cluster, md v1alpha1.MachineDeployment) (*template.ParamsMainAutoScalingGroup, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	minDesiredNodes := minDesiredWorkers(key.WorkerScalingMin(md), key.WorkerScalingMax(md), cc.Status.TenantCluster.TCCP.ASG.DesiredCapacity)
+
+	autoScalingGroup := &template.ParamsMainAutoScalingGroup{
+		AvailabilityZones: key.WorkerAvailabilityZones(md),
+		Cluster: template.ParamsMainAutoScalingGroupCluster{
+			ID: key.ClusterID(&md),
+		},
+		DesiredCapacity:       minDesiredNodes,
+		MaxBatchSize:          workerCountRatio(minDesiredNodes, 0.3),
+		MaxSize:               key.WorkerScalingMax(md),
+		MinInstancesInService: workerCountRatio(minDesiredNodes, 0.7),
+		MinSize:               key.WorkerScalingMin(md),
+		Name:                  key.MachineDeploymentASGName(&md),
+		Subnets:               nil, // TODO
```

Comment:
There are several TODOs here now which we need to address in upcoming PRs. I am happy for now to got this mess compiled. 🙈 hashtagbabysteps